turbopy/core.py

```diff
@@ -611,15 +632,25 @@ def __init__(self, owner: Simulation, input_data: dict):
     def inspect_resource(self, resource: dict):
         """Save references to data from other PhysicsModules
 
-        If your subclass needs the data described by the key, now's their chance to
-        save a reference to the data
+        If your subclass needs the data described by the key, now's their
+        chance to save a reference to the data
+
+        Parameters
+        ----------
+        resource: dict
+            ????
         """
         pass
 
     def diagnose(self):
         """Perform diagnostic step
 
         This gets called on every step of the main simulation loop.
+
+        Raises
+        ------
+        NotImplementedError
+            Method or function hasn't been implemented yet
```

Comment:
I think we should add a note saying _why_ it raises this error. In this case, it's because this class is an abstract base class. Which means that this function `diagnose` is one of the functions that derived classes must implement in order to be a "concrete" class.

Reply:
I'll go ahead and write that, and then I think this is ready for merging.